src/librustc_middle/mir/interpret/error.rs

```diff
@@ -450,13 +450,13 @@ pub enum UnsupportedOpInfo {
     ReadForeignStatic(DefId),
     /// Could not find MIR for a function.
     NoMirFor(DefId),
+    /// Encountered raw bytes where we needed a pointer.
+    ReadBytesAsPointer,
 
     // The variants below are only reachable from CTFE/const prop, miri will never emit them.
 
     /// Encountered a pointer where we needed raw bytes.
     ReadPointerAsBytes,
```

Comment:
This one needs to be moved up, too.

Reply:
how can that fail? I thought intptrcast can convert any `Pointer` to an integer?

Reply:
This gets raised e.g. when doing a `u8` load of a single byte of a pointer.

Reply:
Actually I think `ReadBytesAsPointer` can be moved *below* the line, that one really should be impossible.

Reply:
Also could you update the doc comment above the enum here, at least remove the part about the fall-back?